tls/client-tls-nonblocking.c

```diff
@@ -33,37 +33,92 @@
 /* wolfSSL */
 #include <wolfssl/options.h>
 #include <wolfssl/ssl.h>
-#include <wolfssl/test.h>
 
 #define DEFAULT_PORT 11111
 
+#define CONNECT_WAIT_MS 2
+
 #define CERT_FILE "../certs/ca-cert.pem"
 
+enum {
+    TEST_SELECT_FAIL,
+    TEST_TIMEOUT,
+    TEST_RECV_READY,
+    TEST_SEND_READY,
+    TEST_ERROR_READY
+};
+
+
+static WC_INLINE int tcp_select(SOCKET_T socketfd, int to_sec)
+{
+    int rx = 1;
+    fd_set fds, errfds;
+    fd_set* recvfds = NULL;
+    fd_set* sendfds = NULL;
+    SOCKET_T nfds = socketfd + 1;
+#if !defined(__INTEGRITY)
+    struct timeval timeout = {(to_sec > 0) ? to_sec : 0, 0};
+#else
+    struct timeval timeout;
+#endif
+    int result;
+
+    FD_ZERO(&fds);
+    FD_SET(socketfd, &fds);
+    FD_ZERO(&errfds);
+    FD_SET(socketfd, &errfds);
+
+    if (rx)
+        recvfds = &fds;
+    else
+        sendfds = &fds;
+
+#if defined(__INTEGRITY)
+    timeout.tv_sec = (long long)(to_sec > 0) ? to_sec : 0, 0;
+#endif
+    result = select(nfds, recvfds, sendfds, &errfds, &timeout);
+
+    if (result == 0)
+        return TEST_TIMEOUT;
+    else if (result > 0) {
+        if (FD_ISSET(socketfd, &fds)) {
+            if (rx)
+                return TEST_RECV_READY;
+            else
+                return TEST_SEND_READY;
+        }
+        else if(FD_ISSET(socketfd, &errfds))
+            return TEST_ERROR_READY;
+    }
+
+    return TEST_SELECT_FAIL;
+}
 
 static int NonBlockingSSL_Shutdown(WOLFSSL* ssl)
 {
-    int error, status = SSL_SHUTDOWN_NOT_DONE;
+    int error, status = WOLFSSL_SHUTDOWN_NOT_DONE;
 
     do {
         error = 0;
-        if (tcp_select(SSL_get_fd(ssl), 2) == TEST_RECV_READY) {
+        if (tcp_select(wolfSSL_get_fd(ssl), CONNECT_WAIT_MS) == TEST_RECV_READY){
             status = wolfSSL_shutdown(ssl);    /* bidirectional shutdown */
             if (status == WOLFSSL_SUCCESS)
                 printf("Bidirectional shutdown complete\n");
             else {
                 error = wolfSSL_get_error(ssl, 0);
             }
         }
-    } while (status == SSL_SHUTDOWN_NOT_DONE || error == SSL_ERROR_WANT_READ ||
-             error == SSL_ERROR_WANT_WRITE);
+    } while (status == WOLFSSL_SHUTDOWN_NOT_DONE || error == WOLFSSL_ERROR_WANT_READ ||
+             error == WOLFSSL_ERROR_WANT_WRITE);
 
     return status;
 }
 
 
 int main(int argc, char** argv)
 {
-    int                sockfd;
+    int                ret;
+    int                sockfd; 
```

Comment:
This `sockfd` needs to be initialized with `SOCKET_INVALID` and checked below on free.